src/java.base/share/classes/sun/security/ssl/SignatureScheme.java

```diff
@@ -488,10 +493,7 @@ static Map.Entry<SignatureScheme, Signature> getSignerOfPreferableAlgorithm(
                     ss.isPermitted(constraints)) {
                 if ((ss.namedGroup != null) && (ss.namedGroup.spec ==
                         NamedGroupSpec.NAMED_GROUP_ECDHE)) {
-                    ECParameterSpec params =
-                            x509Possession.getECParameterSpec();
-                    if (params != null &&
-                            ss.namedGroup == NamedGroup.valueOf(params)) {
```

Comment:
look at here.
if : param == null and namedGroup == null and ss.namedGroup == null,
the codes before this change will not enter the if branch, but the codes after this changes will enter the if branch.

Reply:
ss.namedGroup should not be null here. The previous if statement already checks that:
```
if ((ss.namedGroup != null) && (ss.namedGroup.spec ==
        NamedGroupSpec.NAMED_GROUP_ECDHE)) {
```